todo_app/app.py

```diff
@@ -11,34 +14,88 @@ def create_app():
     config = Config()
     app.config.from_object(config)
 
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        github_login = "https://github.com/login/oauth/authorize?client_id=" + config.CLIENT_ID
+        return redirect(github_login)
+
+    @login_manager.user_loader
+    def load_user(user_id):
+        return User(user_id)
+
+    login_manager.init_app(app)
+
     @app.route('/')
+    @login_required
     def index():
         items = get_items(config)
         items = sorted(items, key=lambda item: item.get_ordering_index())
         item_view_model = ViewModel(items)
-        return render_template("index.html", item_view_model=item_view_model)
+        isWriter = current_user.is_anonymous or current_user.role == UserRoles.WRITER
```

Comment:
Minor, but I'd be tempted to wrap this into a function in the same file you've generated the `requireWriter` decorator; it feels like there's some fairly specific logic here that doesn't necessarily need to be here